src/sentry/scripts/similarity/index.lua

```diff
@@ -30,28 +30,55 @@ if not pcall(redis.replicate_commands) then
     redis.log(redis.LOG_DEBUG, 'Could not enable script effects replication.')
 end
 
-local function identity(value)
-    return value
+
+-- Utilities
+
+local function identity(...)
+    return ...
 end
 
-local function range(start, stop)
-    local result = {}
-    for i = start, stop do
-        table.insert(result, i)
+local function sum(t)
+    return table.ireduce(
+        t,
+        function (total, value)
+            return total + value
+        end,
+        0
+    )
+end
+
+local function avg(t)
+    return sum(t) / #t
+end
+
+function table.count(t)
+    -- Shitty O(N) table size implementation
+    local n = 0
+    for k in pairs(t) do
+        n = n + 1
     end
-    return result
+    return n
 end
 
-function table.ifilter(t, f)
+function table.trim(t, n)
     local result = {}
-    for i, value in ipairs(t) do
-        if f(value) then
-            table.insert(result, value)
-        end
+    for i = 1, n do
+        result[i] = t[i]
     end
     return result
 end
 
+function table.get_or_set_default(t, k, f)
+    local v = t[k]
+    if v ~= nil then
+        return v
+    else
+        v = f(k)
+        t[k] = v
+        return v
+    end
+end
+
 function table.imap(t, f)
```

Comment:
I'm likely going to change `imap` and `izip` to just be iterator based to save memory.